### DISCUSSION TITLE
How to trigger a hook so a field is updated on "create" action?

### DISCUSSION THREAD

**Discussion by JonL1** [directus/directus#12322] [Archive]
I am using nanoid to generate short ids. Every table has a field to store it. 
What would be the best approach to generate a nanoid and save it to the record when saving the form to create a record?
If anyone could provide some high level pseudo I can go on from there.

**Comment by rijkvanzanten** [comment]
Assuming your column is named `shortid`, it will be something like this (haven't tested it, but should be enough to get you started):

```js
const nanoid = require('nanoid');

module.exports = function registerHooks({ filter }) {
  filter('items.create', (payload) => {
    if (Array.isArray(payload)) return payload.map((item) => item.shortid = nanoid());

    payload.shortid = nanoid();
    return payload;
  });
}
```

**Reply by NilsBaumgartner1994** [reply]
How to import nanoid now in an extension?
